src/transformState.js

```diff
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const [key, value] of Object.entries(action.extraData)) {
+          state[key] = value;
+        }
+    }
+
+    switch (action.type) {
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete state[key];
+        }
+    }
+
+    switch (action.type) {
+      case 'clear':
+        for (const key of Object.keys(state)) {
+          delete state[key];
+        }
+    }
+  }
```

Comment:
thats not how switch works

`
swith(action.type){
  case 'addProperties': ...
  case 'removeProperties': ...
  default: ...
}
`

https://mate.academy/learn/java-basics-extended/java-switch-case#/theory